Вариант 3/Вариант 3/Program.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections.Generic;
+
+class Program
+{
+   
+    static List<int> Eratosthenes(int n)
+    {
+        bool[] sieve = new bool[n + 1];
+        List<int> primes = new List<int>();
+
+       
+        for (int i = 2; i <= n; i++)
+            sieve[i] = true;
+
+        
+        for (int i = 2; i * i <= n; i++)
+        {
+            if (sieve[i])
+            {
+                for (int j = i * i; j <= n; j += i)
+                {
+                    sieve[j] = false;
+                }
+            }
+        }
+
+        
+        for (int i = 2; i <= n; i++)
+        {
+            if (sieve[i])
+            {
+                primes.Add(i);
+            }
+        }
+
+        return primes;
+    }
```

Comment:
Лишние скобки затрудняют чтение кода. Не только здесь, но и в других методах.
Лишние пустые строки тоже.
```suggestion
    static List<int> Eratosthenes(int n)
    {
        bool[] sieve = new bool[n + 1];
        List<int> primes = new List<int>();

        for (int i = 2; i <= n; i++)
            sieve[i] = true;

        for (int i = 2; i * i <= n; i++)
            if (sieve[i])
                for (int j = i * i; j <= n; j += i)
                    sieve[j] = false;
        
        for (int i = 2; i <= n; i++)
            if (sieve[i])
                primes.Add(i);

        return primes;
    }
```